src/client.ts

```diff
@@ -394,20 +400,34 @@ export class RSPClient {
      * @param timeout timeout in milliseconds
      */
     removeDeployable(req: Protocol.ModifyDeployableRequest, timeout: number = 60000): Promise<Protocol.Status> {
-        return this.serverUtil.removeDeployable(req, timeout);
+        return this.publishingUtil.removeDeployable(req, timeout);
     }
 
     /**
-     * Publish a given server
+     * Publishes a given server
      *
      * @param server A server handle see {@link Protocol.ServerHandle}
      * @param timeout timeout in milliseconds
      */
     publish(req: Protocol.PublishServerRequest, timeout: number = 60000): Promise<Protocol.Status> {
-        return this.serverUtil.publish(req, timeout);
+        return this.publishingUtil.publish(req, timeout);
     }
 
     /**
+     * Lists the downloadable runtimes
+     */
+    listDownloadRuntimes(timeout?: number): Promise<Protocol.ListDownloadRuntimeResponse> {
+      return this.downloadRuntimes.listDownloadableRuntimes(timeout);
+  }
+
+    /**
+     * Starts the workflow to download a runtime
+     */
+    downloadRuntime(req: Protocol.DownloadSingleRuntimeRequest, timeout?: number): Promise<Protocol.WorkflowResponse> {
```

Comment:
Please add a basic call check to client-test for each new method. The other checks there should provide a nice template.

Reply:
the test for #downloadRuntime is here: https://github.com/redhat-developer/rsp-client/pull/68/files#diff-4bf5fb55a09c699ba601b00188e0ef58R135

What am I missing?

Reply:
What I meant is to add a test if client.downloadRuntime calls downloadRuntimes.downloadRuntime correctly (analogically for the other method).

It's a quick check on the API level, like everything in this file:
https://github.com/redhat-developer/rsp-client/blob/master/test/client-test.ts

Reply:
please point to what is missing. I don't see any difference to other tests like ex. serverModel-test.ts, What am I missing?

Imho the fact that the utility class is being called is something that we shouldn't assert. The way the code is organized - for instance not having the code in the client class but in utils class - is an internal implementation detail that we should not assert. On the other hand, the request is API and should be asserted.

Reply:
Why should we not assert that an API method does what we want it to? Even if in this case it is just to call another method.

Reply:
still, the levels of indirections are an implementation detail, they have to effect on the functionality/behaviour and as such shouldn't be asserted.